src/ripple/app/ledger/LedgerTiming.h

```diff
@@ -50,6 +51,78 @@ roundCloseTime (
     NetClock::time_point closeTime,
     NetClock::duration closeResolution);
 
+/** Determines whether the current ledger should close at this time.
+
+    This function should be called when a ledger is open and there is no close
+    in progress, or when a transaction is received and no close is in progress.
+
+    @param anyTransactions indicates whether any transactions have been received
+    @param previousProposers proposers in the last closing
+    @param proposersClosed proposers who have currently closed this ledger
+    @param proposersValidated proposers who have validated the last closed
+                              ledger
+    @param previousMSeconds time, in milliseconds, for the previous ledger to
+                            reach consensus (in milliseconds)
+    @param currentMSeconds time, in milliseconds, since the previous ledger's
+                           (possibly rounded) close time
+    @param openMSeconds time, in milliseconds, waiting to close this ledger
+    @param idleInterval the network's desired idle interval
+*/
+bool
+shouldCloseLedger (
+    bool anyTransactions,
+    int previousProposers,
+    int proposersClosed,
+    int proposersValidated,
+    std::chrono::milliseconds previousMSeconds,
```

Comment:
While you're moving stuff, please consider giving better names for these. The `MSeconds` is no longer needed now that the variable type is meaningful. If you do, please update the comment as well.